apps/main/src/llamalend/features/market-list/filters/RangeSliderFilter.tsx

```diff
@@ -42,30 +69,75 @@ export const RangeSliderFilter = <T,>({
   const step = useMemo(() => Math.ceil(+maxValue.toPrecision(2) / 100), [maxValue])
-  const defaultValue = useMemo((): NumberRange => {
+  const isPowerScale = scale === 'power'
+  /** Find a power exponent that feels "good" based on the max value  */
+  const powerExponent = useMemo(() => {
+    if (!isPowerScale || maxValue <= minValue) return undefined
+    const safeMax = Math.max(maxValue, 1)
+    const exponent = Math.trunc(Math.log10(safeMax) - 2)
+    return exponent > 1 ? exponent : 1
+  }, [isPowerScale, maxValue, minValue])
+
+  const canUsePowerScale = isPowerScale && powerExponent != null
+
+  /** Helpers that convert real values to and from the slider's value space. */
+  const sliderValueTransform = useMemo(() => {
+    if (!canUsePowerScale) {
+      return undefined
+    }
+    const [sliderMin, sliderMax] = [0, 1]
+
+    /** Convert the current value into the 0-1 slider space. */
+    const toSlider = (value: number) =>
+      invertPowerMap(clamp(value, minValue, maxValue), minValue, maxValue, powerExponent)
+    /** Bring the normalized slider value back into the actual units. */
+    const fromSlider = (value: number) =>
+      powerMap(clamp(value, sliderMin, sliderMax), minValue, maxValue, powerExponent)
+
+    return {
+      toSlider,
+      fromSlider,
+      sliderMin,
+      sliderMax,
+      sliderStep: 0.001,
+    }
+  }, [canUsePowerScale, powerExponent, minValue, maxValue])
+
+  const defaultRange = useMemo<NumberRange>(() => [defaultMinimum, maxValue], [defaultMinimum, maxValue])
+  // Currently applied filter range
+  const appliedRange = useMemo((): NumberRange => {
     const [min, max] = (columnFilters[id] as NumberRange) ?? []
     return [min ?? defaultMinimum, max ?? maxValue]
   }, [columnFilters, id, maxValue, defaultMinimum])
-  const isMobile = useIsMobile()
 
   const [range, setRange] = useUniqueDebounce({
-    defaultValue,
+    // Separate default and applied range, because the input's onBlur event that didn’t actually change anything could trigger the callback, and would clear the filter.
+    defaultValue: appliedRange,
     callback: useCallback(
       (newRange: NumberRange) =>
         setColumnFilter(
           id,
-          newRange.every((value, i) => value === defaultValue[i])
-            ? undefined // remove the filter if the range is the same as the default
+          newRange.every((value, i) => value === defaultRange[i])
+            ? undefined // remove the filter if the range is the same as the default range
             : [newRange[0] === defaultMinimum ? null : newRange[0], newRange[1] === maxValue ? null : newRange[1]],
         ),
-      [defaultMinimum, defaultValue, id, maxValue, setColumnFilter],
+      [defaultMinimum, defaultRange, id, maxValue, setColumnFilter],
     ),
   })
 
-  const onChange = useCallback<OnSliderChange>((_, newRange) => setRange(newRange as NumberRange), [setRange])
+  const onChange = useCallback<OnSliderChange>(
```

Comment:
I think we can easily inline this in `onChange={onChange}` if all it does is just calling `setRange(newRange as NumberRange)`